packages/core/src/scope.ts

```diff
@@ -135,11 +129,12 @@ export class Scope implements BaseScope {
   /**
    * Sets the breadcrumbs in the scope
    * @param breadcrumbs
+   * @param maxBreadcrumbs
    */
-  public addBreadcrumb(breadcrumb: Breadcrumb, maxBreadcrumbs: number): void {
-    this.breadcrumbs = [...this.breadcrumbs, breadcrumb].slice(
-      -Math.max(0, maxBreadcrumbs),
-    );
+  public addBreadcrumb(breadcrumb: Breadcrumb, maxBreadcrumbs?: number): void {
+    this.breadcrumbs = maxBreadcrumbs
```

Comment:
_nit_: `maxBreadcrumbs > 0`, after which you can remove `Math.max` down below. Alternatively, use a default value in the parameter list (`maxBreadcrumbs: number = 0`)